### PR TITLE
Adds Schedule Timestep Advancement Logging

### PR DESCRIPTION
**Pull request by wito**
This feature adds logging of the simulation time step in `Schedule.cpp` as `DATES` and `TSTEP` keywords are encountered.

### CODE REVIEW

**Comment by wito** [comment]
jenkins build this please

File: src/opm/parser/eclipse/EclipseState/Schedule/Schedule.cpp [hunk 1]
```diff
@@ -342,13 +344,26 @@ namespace {
             }
 
             else if (keyword.name() == "DATES") {
+                for (const auto& record : keyword) {
+                    const std::time_t date = TimeMap::timeFromEclipse(record);
+                    char dateString[20];
```
Comment (joakim-hove) [inline src/opm/parser/eclipse/EclipseState/Schedule/Schedule.cpp:349]:
```
const std::time_t date = this->simTime( context->currentStep );
auto ts = TimeStampUTC( date );
OpmLog::info( fmt::format("Processing simulation date {:02d}/{:02d}/{}" , ts.day(),ts.month(), ts.year()));
```

**Comment by atgeirr** [comment]
Sounds useful. Could you post a little of the output in context?

**Comment by wito** [comment]
> Sounds useful. Could you post a little of the output in context?

It looks something like this:

```
Info: Processing simulation date 2019-08-01 (18109.0 DAYS).
Info: SHUT well: INJ1 at report step: 17
```

**Comment by wito** [comment]
I'm having an issue with the current code that I don't really understand the codebase well enough to begin grappling with; the `simTime()` method is crashing the simulation when it has been restarted, and I don't know how to fix that.

**Comment by joakim-hove** [comment]
> Processing simulation date 2019-08-01 (18109.0 DAYS).

Should also include the report step somewhere in that string.

**Comment by joakim-hove** [comment]
> I'm having an issue with the current code that I don't really understand the codebase well enough to begin grappling with; the simTime() method is crashing the simulation when it has been restarted, and I don't know how to fix that.

I'll call you - but a little more details in the meantime?

**Comment by wito** [comment]
Pertinent info from the test output:

```
18: Advancing simulation by 59 steps.
18: An error occured while creating the reservoir schedule
18: 
18: libc++abi.dylib: terminating with uncaught exception of type std::invalid_argument: Tried to get time information from the base case in restarted run
```

File: opm/parser/eclipse/EclipseState/Schedule/Schedule.hpp [hunk 2]
```diff
@@ -442,6 +442,9 @@ namespace Opm
             }
         }
 
+        std::string simulationDate(std::size_t currentStep) const;
```
Comment (joakim-hove) [inline opm/parser/eclipse/EclipseState/Schedule/Schedule.hpp:445]: These small utilities do not deserve to become members in the already crowded Schedule class - just make anonymous free functions.

**Comment by joakim-hove** [comment]
This is rebased and continued here: #1971

**Pull request closed without merging**